chain/jsonrpc/src/lib.rs

```diff
@@ -725,54 +774,50 @@ impl JsonRpcHandler {
+        request: near_jsonrpc_primitives::types::changes::RpcStateChangesRequest,
+    ) -> Result<
+        near_jsonrpc_primitives::types::changes::RpcStateChangesInBlockResponse,
+        near_jsonrpc_primitives::types::changes::RpcStateChangesError,
+    > {
+        let block = self.view_client_addr.send(GetBlock(request.block_reference.into())).await??;
 
         let block_hash = block.header.hash.clone();
-        jsonify(self.view_client_addr.send(GetStateChangesInBlock { block_hash }).await.map(|v| {
-            v.map(|changes| RpcStateChangesInBlockResponse {
-                block_hash: block.header.hash,
-                changes,
+
+        Ok(self.view_client_addr.send(GetStateChangesInBlock { block_hash }).await.map(|v| {
+            v.map(|changes| {
+                near_jsonrpc_primitives::types::changes::RpcStateChangesInBlockResponse {
+                    block_hash: block.header.hash,
+                    changes,
+                }
             })
-        }))
+        })??)
     }
 
-    async fn changes_in_block_by_type(&self, params: Option<Value>) -> Result<Value, RpcError> {
-        let RpcStateChangesRequest { block_reference, state_changes_request } =
-            parse_params(params)?;
-        // TODO refactor it. Changed to keep it working before refactoring
-        let result = self.view_client_addr.send(GetBlock(block_reference)).await?;
-        let block = match result {
-            Ok(block) => block,
-            Err(err) => {
-                return Err(RpcError::from(
-                    near_jsonrpc_primitives::types::blocks::RpcBlockError::from(err),
-                ))
-            }
-        };
+    async fn changes_in_block_by_type(
+        &self,
+        request: near_jsonrpc_primitives::types::changes::RpcStateChangesRequest,
+    ) -> Result<
+        near_jsonrpc_primitives::types::changes::RpcStateChangesResponse,
+        near_jsonrpc_primitives::types::changes::RpcStateChangesError,
+    > {
+        let block = self.view_client_addr.send(GetBlock(request.block_reference.into())).await??;
 
         let block_hash = block.header.hash.clone();
-        jsonify(
-            self.view_client_addr
-                .send(GetStateChanges { block_hash, state_changes_request })
-                .await
-                .map(|v| {
-                    v.map(|changes| RpcStateChangesResponse {
-                        block_hash: block.header.hash,
-                        changes,
-                    })
-                }),
-        )
+        Ok(self
+            .view_client_addr
+            .send(GetStateChanges {
+                block_hash,
+                state_changes_request: request.state_changes_request,
+            })
+            .await
+            .map(|v| {
+                v.map(|changes| near_jsonrpc_primitives::types::changes::RpcStateChangesResponse {
+                    block_hash: block.header.hash,
+                    changes,
+                })
+            })??)
```

Comment:
Those maps were unfortunate since `jsonify` wanted to have `Result<Result<...>>`, but now we can refactor it into:

```suggestion
        let changes = self
            .view_client_addr
            .send(GetStateChanges {
                block_hash,
                state_changes_request: request.state_changes_request,
            })
            .await??;
        Ok(near_jsonrpc_primitives::types::changes::RpcStateChangesResponse {
            block_hash: block.header.hash,
            changes,
        })
```